packages/flutter/test/material/dialog_theme_test.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io' show Platform;
+
+import 'package:flutter/material.dart';
+import 'package:flutter/src/material/dialog_theme.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+MaterialApp _appWithAlertDialog(WidgetTester tester, AlertDialog dialog, {ThemeData theme}) {
+  return MaterialApp(
+    theme: theme,
+    home: Material(
+        child: Builder(
+            builder: (BuildContext context) {
+              return Center(
+                  child: RaisedButton(
+                      child: const Text('X'),
+                      onPressed: () {
+                        showDialog<void>(
+                          context: context,
+                          builder: (BuildContext context) {
+                            return RepaintBoundary(key: _painterKey, child: dialog);
+                          },
+                        );
+                      }
+                  )
+              );
+            }
+        )
+    ),
+  );
+}
+
+final Key _painterKey = UniqueKey();
+
+void main() {
+  testWidgets('Custom dialog shape', (WidgetTester tester) async {
+    const RoundedRectangleBorder customBorder =
+      RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(16.0)));
+    const AlertDialog dialog = AlertDialog(
+      title: Text('Title'),
+      actions: <Widget>[ ],
+    );
+    final ThemeData theme = ThemeData(dialogTheme: const DialogTheme(shape: customBorder));
+
+    await tester.pumpWidget(
+      _appWithAlertDialog(tester, dialog, theme: theme)
+    );
+    await tester.tap(find.text('X'));
+    await tester.pump(); // start animation
```

Comment:
Just `await tester.pumpAndSettle()` should be sufficient.

Reply:
Done.